network/network.go

```diff
@@ -213,47 +213,38 @@ func (n *Network) Configure(config core.ServerConfig) error {
 	}
 
 	// register callback from DAG to other engines, with payload only.
-	n.state.RegisterPayloadObserver(n.emitEvents, true)
-
-	// register observers to publish to other engines. Non-transactional, so will be published to other engines at most once.
-	n.state.RegisterTransactionObserver(func(_ stoabs.WriteTx, transaction dag.Transaction) error {
-		n.publish(TransactionAddedEvent, transaction, nil)
-		return nil
-	}, false)
-	n.state.RegisterPayloadObserver(func(transaction dag.Transaction, payload []byte) error {
-		n.publish(TransactionPayloadAddedEvent, transaction, payload)
-		return nil
-	}, false)
+	if _, err = n.state.Notifier("nats", n.emitEvents,
+		dag.WithPersistency(dagStore),
+		dag.WithSelectionFilter(func(event dag.Event) bool {
+			return event.Type == dag.PayloadEventType
+		})); err != nil {
+		return err
+	}
 
 	return nil
 }
 
-// emitEvents is called when a transaction is being added to the DAG.
-// It runs within the transactional context because if the event fails, the transaction must also fail.
-// If the transaction fails for some reason (storage) then the event is still emitted. This is ok because the transaction was already validated.
-// Most likely the transaction will be added at a later stage and another event will be emitted.
-// It only emits events when both the payload and transaction are present. This is the case from both state.Add and from state.WritePayload.
-func (n *Network) emitEvents(tx dag.Transaction, payload []byte) error {
-	if tx != nil && payload != nil {
-		_, js, err := n.eventPublisher.Pool().Acquire(context.Background())
-		if err != nil {
-			return fmt.Errorf(errEventFailedMsg, err)
-		}
+// emitEvents is called when a payload is added.
+func (n *Network) emitEvents(event dag.Event) (bool, error) {
+	_, js, err := n.eventPublisher.Pool().Acquire(context.Background())
+	if err != nil {
+		return false, fmt.Errorf(errEventFailedMsg, err)
+	}
 
-		twp := events.TransactionWithPayload{
-			Transaction: tx,
-			Payload:     payload,
-		}
-		twpData, err := json.Marshal(twp)
-		if err != nil {
-			return fmt.Errorf(errEventFailedMsg, err)
-		}
+	twp := events.TransactionWithPayload{
+		Transaction: event.Transaction,
+		Payload:     event.Payload,
+	}
+	twpData, err := json.Marshal(twp)
+	if err != nil {
+		return false, fmt.Errorf(errEventFailedMsg, err)
+	}
 
-		if _, err = js.PublishAsync(events.TransactionsSubject, twpData); err != nil {
-			return fmt.Errorf(errEventFailedMsg, err)
-		}
+	if _, err = js.PublishAsync(events.TransactionsSubject, twpData); err != nil {
```

Comment:
the subject can use some rewording (or at least the Godoc):

```
// TransactionsSubject defines the NATS subject used for transactions
// Payload: TransactionWithPayload
const TransactionsSubject = "TRANSACTIONS.tx"
```

Reply:
not part of this PR